cpp/include/Ice/IconvStringConverter.h

```diff
@@ -89,21 +90,60 @@ class IconvStringConverter : public IceUtil::BasicStringConverter<charT>
 
     IconvStringConverter(const std::string&);
 
-    virtual ~IconvStringConverter();
-
     virtual Ice::Byte* toUTF8(const charT*, const charT*, Ice::UTF8Buffer&) const;
 
     virtual void fromUTF8(const Ice::Byte*, const Ice::Byte*, std::basic_string<charT>&) const;
 
 private:
 
+    struct DescriptorHolder
+    {
+        std::pair<iconv_t, iconv_t> descriptor;
+
+        DescriptorHolder() :
+            descriptor(iconv_t(0), iconv_t(0))
+        {
+        }
+
+        DescriptorHolder(std::pair<iconv_t, iconv_t> desc) :
+            descriptor(desc)
+        {
+        }
+
+        DescriptorHolder(const DescriptorHolder& other) = delete;
+        DescriptorHolder& operator=(const DescriptorHolder& other) = delete;
+
+        DescriptorHolder(DescriptorHolder&& other) noexcept = delete;
+
+        DescriptorHolder& operator=(DescriptorHolder&& other)
+        {
+            if (this != &other)
+            {
+                descriptor = std::move(other.descriptor);
```

Comment:
This assumes that iconv_t is movable?

I think it would be cleaner to not make this assumption and simply copy-assign. 
Or even better, remove all these functions - see below.